docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+the analysis of near real time and past logs, from a central endpoint. As such
+these logs should be stored on the platform, for a configurable period.
+Browsing the logs is accessible through an API and a UI. The UI should ease
+correlation between logs and health/performance KPIs as well as alerts.
+
+For organisations having their own Log centralisation system (like Splunk or
+Elasticsearch), MetalK8s should provide a way to push logs into it.
+
+The following requirements focus on application logs. Audit logs are not part
+of the requirements.
+
+Requirements
+------------
+
+*Lightweight*
+
+A lightweight tool to store and expose logs is required in order to minimize
+the HW footprints (CPU, RAM, Disks):
+
+- limited history: Storing the logs for very large period (3 years) is not
+  something metalK8s needs to provide as a feature. This can be achieved using
+  external log centralisation system.
+
+- stable ingestion: It is important to guaranty stable ingestion of the logs
+  and less important to guaranty stable performances when browsing/searching
+  the logs. However, peak loads related to complex logs queries should not
+  impact the application workloads and deploying log storage and search service
+  on infra nodes might help achieving this isolation.
+
+- stream indexing: It is not required to have automatic indexing of logs
+  content. Instead, the log centralisation service should offer basic features
+  to group/filter logs per tag/metadata defining the log stream.
+
+*Accessible from a central UI/API*
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as journal logs, including (kubelet, containerd, salt-minion
+, kernel, initrd, services, etc ...) in Grafana. One can correlate logs and
+metrics or alerts in one single Grafana Dashboard.
+Browsing logs can be achieved through a documented API in order to expose logs
+in MetalK8s UIs or other workloads UIs if needed.
+
+*Persistence, Retention*
+
+Logs should be stored on a persistent storage. Platform Administrator can
+configure a max retention period. Some automatic purging mechanism is triggered
+when logs are older than the retention period or when the persistent storage is
+about to reach its capacity limit.
+A typical and default retention period is of 2 weeks.
+
+*Horizontally scalable (capacity and performance)*
+
+The Platform Administrator can add nodes or disks in order to increase storage
+capacity for logs (to keep more history or to store more workloads logs) or to
+increase performances for ingesting or querying the logs. It is also possible
+to fine tune the deployment configuration in order to have more replicas and
+ideally the Log centralisation service supports horizontal POD autoscaling.
+
+*Highly Available*
+
+Log collection, storage and browsing services can be replicated in order to
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
```

Comment:
In this case, of HA with replicas everywhere, then you can *not* add a new node to extend the storage (as you mentioned in the previous section)

Reply:
@TeddyAndrieux right ... so how it works with Prometheus? extending the storage means extending PV size?

Reply:
Yes, sure both doable but not at the same time and I think if we keep these 2 requirements we need to say it somewhere in this document.
Anyway as first iteration we will likely do the HA one as it's the simplest IMO

For prometheus (and it will be the same for Log centralization) a lot of possibility but pros and cons for each ....
"on pourra pas sauver tout le monde" :smiley:

Reply:
Okay actually the requirement is about being able to scale the service in order ingest and store more logs

Reply:
My point is that you cannot match the 2 requirements in a deployment, either it's horizontally scalable either it's HA (with lose all but one).
And btw horizontally scalable need more logic on our side.

Anyway we can have the ability to handle both different deployment (likely with some CSC tunning) but a single deployment cannot match both. 

To me, it should be mentioned somewhere in this document and if we really want to support both then you should add a point about it in the `Deployment & Configuration` section